src/main/resources/lib/nextjs/postprocessing.es6

```diff
@@ -1,15 +1,40 @@
 /** Replace URL refs in both HTML, JS and JSON sources from pointing to frontend-urls to making them sub-urls below the extFrontendProxy service */
-export const getBodyWithReplacedUrls = (req, body, proxyUrlWithSlash, isCss) => {
+import {getFrontendServerUrl, removeEndSlashPattern} from "./connection-config";
+
+export const getBodyWithReplacedUrls = (req, body, proxyUrlWithSlash, isCss, config) => {
+    const wSpaces = '[ \\r\\n\\t]*';
     // double slashes is the right way of escaping bracket!
-    const cssUrlPattern = new RegExp(`url\\([ \r\n\t]*(['"\`])?\/([^'"\`]*)['"\`]?[ \r\n\t]*\\)`, "g");
+    const cssUrlPattern = new RegExp(`url\\(${wSpaces}(['"\`])?\/([^'"\`]*)['"\`]?${wSpaces}\\)`, "g");
 
     // Replace local absolute root URLs (e.g. "/_next/..., "/api/... etc)
-    const nextApiPattern = new RegExp(`(['"\`])([^'"\` \n\r\t]*\/)((?:_next(?!\/image?)\/|api\/)[^'"\` \n\r\t]*)['"\`]`, "g");
+    const nextApiPattern = new RegExp(`(URL\\(${wSpaces})?(['"\`])([^'"\` \n\r\t]*\/)((?:_next\/(?!image)|api\/)[^'"\` \n\r\t]*)['"\`]`, "gmi");
+
+    const getFullUrl = (substring, isUrlConstructor, linkDomain, linkLocation, frontendDomain, proxyUrlWithSlash) => {
+        let result;
+        if (isUrlConstructor || linkDomain.replace(removeEndSlashPattern, '') === frontendDomain) {
+            // keep the link, as it is an url constructor (can't be relative) or has absolute direct url to frontend
+            result = linkDomain;
+        } else {
+            result = proxyUrlWithSlash;
+        }
+
+        return result + linkLocation;
+    }
+
+    const frontendServerUrl = getFrontendServerUrl(config);
+    const replacerFn = (substring, g1, g2, g3, g4) => {
+        // groups:
+        // 1 - url(
+        // 2 - "
+        // 3 - domain
+        // 4 - location
+        const result = `${g2}${getFullUrl(substring, !!g1, g3, g4, frontendServerUrl, proxyUrlWithSlash)}${g2}`;
+        // log.info('Replacing ' + substring + ' -> ' + result);
+        return result;
+    };
 
     // Don't do next static urls replacement in css
-    const result = isCss ?
-        body :
-        body.replace(nextApiPattern, `$1${proxyUrlWithSlash}$3$1`);
+    const result = isCss ? body : body.replace(nextApiPattern, replacerFn);
```

Comment:
This code is doing a lot of complex transformation which are unnecessary if `isCss==true` (see line 37).

So this needs to be refactored to something like this:
```Typescript
export const getBodyWithReplacedUrls = (req, body, proxyUrlWithSlash, isCss, config) => {
  if (isCss) {
    return body.replace(cssUrlPattern, `url($1${proxyUrlWithSlash}$2$1)`);
  }

   // ... the rest of the logic here ...
}
```
  }

Reply:
Not quite true, because 
- cssUrlPattern is run for every filetype, so should always be executed.
- nextApiPattern is run for every filetype, except css.
So in case of non-css files they should be both executed.

Reply:
This is current code:
```
const wSpaces = '[ \\r\\n\\t]*';
const cssUrlPattern = new RegExp(`url\\(${wSpaces}(['"\`])?\/([^'"\`]*)['"\`]?${wSpaces}\\)`, "g");

/// lots of code here

const result = isCss ? body : body.replace(nextApiPattern, replacerFn);
return result.replace(cssUrlPattern, `url($1${proxyUrlWithSlash}$2$1)`);
```

if `isCss` is true, then `result == body` and the next line is the same as doing `return body.replace(cssUrlPattern, `url($1${proxyUrlWithSlash}$2$1)`);`

Nothing between lines 10 and 37 is needed if `isCss` is true.

So it should be 

```
const wSpaces = '[ \\r\\n\\t]*';
// double slashes is the right way of escaping bracket!
const cssUrlPattern = new RegExp(`url\\(${wSpaces}(['"\`])?\/([^'"\`]*)['"\`]?${wSpaces}\\)`, "g");

if (isCss) {
  return body.replace(cssUrlPattern, `url($1${proxyUrlWithSlash}$2$1)`);
}

 // ... the rest of the logic here ...
```